Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -8642,6 +8658,70 @@ void WebPageProxy::setCursorHiddenUntilMouseMoves(bool hiddenUntilMouseMoves)
     pageClient().setCursorHiddenUntilMouseMoves(hiddenUntilMouseMoves);
 }
 
+void WebPageProxy::mouseEventHandlingCompleted(std::optional<WebEventType> eventType, bool handled)
+{
+    // Retire the last sent event now that WebProcess is done handling it.
+    MESSAGE_CHECK(m_process, !internals().mouseEventQueue.isEmpty());
+    auto event = internals().mouseEventQueue.takeFirst();
+    if (eventType) {
+        MESSAGE_CHECK(m_process, *eventType == event.type());
+#if ENABLE(CONTEXT_MENU_EVENT)
+        if (event.button() == WebMouseEventButton::Right) {
+            if (event.type() == WebEventType::MouseDown) {
+                ASSERT(m_contextMenuPreventionState == EventPreventionState::Waiting);
+                m_contextMenuPreventionState = handled ? EventPreventionState::Prevented : EventPreventionState::Allowed;
+            } else if (m_contextMenuPreventionState != EventPreventionState::Waiting)
+                m_contextMenuPreventionState = EventPreventionState::None;
+
+            processContextMenuCallbacks();
+        }
+#endif
+    }
+
+    if (!internals().mouseEventQueue.isEmpty()) {
+        LOG(MouseHandling, " UIProcess: handling a queued mouse event from didReceiveEvent");
```

Comment:
Nit - does this log message need to be updated? (Since `mouseEventHandlingCompleted` can be called from outside of `didReceiveEvent`)